read-fonts/src/tables/postscript/scale.rs

```diff
@@ -0,0 +1,541 @@
+        assert_eq!(cff.subfont_index(GlyphId::new(1)), 0);
+        assert_eq!(cff.global_subrs().count(), 17);
+        let subfont = cff.subfont(0, 0.0, Default::default(), false).unwrap();
+        let hinting_params = subfont.hint_params;
+        check_blues(
+            &hinting_params.blues,
+            &[
+                (-15.0, 0.0),
+                (536.0, 547.0),
+                (571.0, 582.0),
+                (714.0, 726.0),
+                (760.0, 772.0),
+            ],
+        );
+        check_blues(&hinting_params.other_blues, &[(-255.0, -240.0)]);
+        assert_eq!(hinting_params.blue_scale, Fixed::from_f64(0.05));
+        assert_eq!(hinting_params.blue_fuzz, Fixed::ZERO);
+        assert_eq!(hinting_params.language_group, 0);
+    }
+
+    #[test]
+    fn read_cantarell_vf() {
+        let font = FontRef::new(font_test_data::CANTARELL_VF_TRIMMED).unwrap();
+        let cff = Scaler::new(&font).unwrap();
+        assert!(cff.is_cff2());
+        assert!(cff.top_dict.var_store.is_some());
+        assert!(cff.top_dict.font_dicts.is_some());
+        assert!(cff.top_dict.private_dict_range.is_none());
+        assert!(cff.top_dict.fd_select.is_none());
+        assert_eq!(cff.subfont_count(), 1);
+        assert_eq!(cff.subfont_index(GlyphId::new(1)), 0);
+        assert_eq!(cff.global_subrs().count(), 0);
+        let subfont = cff.subfont(0, 0.0, Default::default(), false).unwrap();
+        let hinting_params = subfont.hint_params;
+        check_blues(
+            &hinting_params.blues,
+            &[(-10.0, 0.0), (482.0, 492.0), (694.0, 704.0), (739.0, 749.0)],
+        );
+        check_blues(&hinting_params.other_blues, &[(-227.0, -217.0)]);
+        assert_eq!(hinting_params.blue_scale, Fixed::from_f64(0.0625));
+        assert_eq!(hinting_params.blue_fuzz, Fixed::ONE);
+        assert_eq!(hinting_params.language_group, 0);
+    }
+
+    #[test]
+    fn read_example_cff2_table() {
+        let cff = Scaler::from_cff2(
+            Cff2::read(FontData::new(font_test_data::cff2::EXAMPLE)).unwrap(),
+            1000,
+        )
+        .unwrap();
+        assert!(cff.is_cff2());
+        assert!(cff.top_dict.var_store.is_some());
+        assert!(cff.top_dict.font_dicts.is_some());
+        assert!(cff.top_dict.private_dict_range.is_none());
+        assert!(cff.top_dict.fd_select.is_none());
+        assert_eq!(cff.subfont_count(), 1);
+        assert_eq!(cff.subfont_index(GlyphId::new(1)), 0);
+        assert_eq!(cff.global_subrs().count(), 0);
+    }
+
+    #[test]
+    fn cantarell_vf_outlines() {
+        compare_glyphs(
+            font_test_data::CANTARELL_VF_TRIMMED,
+            font_test_data::CANTARELL_VF_TRIMMED_GLYPHS,
+        );
+    }
+
+    #[test]
+    fn noto_serif_display_outlines() {
+        compare_glyphs(
+            font_test_data::NOTO_SERIF_DISPLAY_TRIMMED,
+            font_test_data::NOTO_SERIF_DISPLAY_TRIMMED_GLYPHS,
+        );
+    }
```

Comment:
I cannot readily tell where the expected values come from when I read the unit test. I assume the answer is freetype, perhaps through naming or comment we could clarify. For example, I can follow this somewhat more easily:

```rust
    #[test]
    fn cff_static_outline_matches_freetype() {
        compare_glyphs(
            font_test_data::NOTO_SERIF_DISPLAY_TRIMMED,
            font_test_data::NOTO_SERIF_DISPLAY_TRIMMED_GLYPHS,
        );
    }

    #[test]
    fn cff2_static_outline_matches_freetype() { ... }
   
   // etc
```

...that's making me realize, it looks like we test nothing more than some outline for each outline format. Shouldn't we build up test assets that use all the drawing commands? E.g. test that a cff with an hstem matches the freetype outline with an hstem, perhaps making a test font with many test glyphs such that each glyph uses "basic" (line, curve, etc) commands plus one unusual (hstem, vstem, etc) command . We could them compare those glyphs for (cff, cff2) x (static, variable).

Reply:
We actually test a number of outlines at multiple sizes and (for VFs) locations in variation space. That was the whole point of adding the extract glyphs script. 

The charstring module has a handcrafted test that contains all operators except hinting. We can’t test those until hinting support lands.

Reply:
Renamed these as suggested. We can increase coverage by adding more glyphs or test fonts to font-test-data. Or additionally, but adjusting the outline instances generated by the extract glyphs script.